Metalama.Framework.Engine/Templating/TemplatingDiagnosticDescriptors.cs

```diff
@@ -508,7 +508,7 @@ internal static readonly DiagnosticDefinition<ISymbol> ExtensionMethodTemplateNo
             = new(
                 "LAMA0271",
                 "Extension method templates are not supported.",
-                "The template '{0}' is an extension method, which is not supported.",
+                "The template '{0}' is an extension method, which is not supported. To introduce an extension method, mark the first parameter of the introduced method with the [This] attribute or programmatically set its IParameterBuilder.IsThis property.",
```

Comment:
It's not clear from the error message _what_ is not supported. I'm not blaming this commit but the previous state. Change the error message to say what exactly is not supported.